src/pages/projectSettings/tabs/NotificationsTab/utils.ts

```diff
@@ -0,0 +1,145 @@
+    case NotificationMethods.EMAIL:
+      return notification.emailInput;
+    case NotificationMethods.WEBHOOK:
+      return notification.webhookInput.urlInput;
+    case NotificationMethods.JIRA_ISSUE:
+      return notification.jiraIssueInput.projectInput;
+    default:
+      return "";
+  }
+};
+
+// Converts the form regexSelector into the proper format for GQL payload.
+// We need to check if the trigger has regex selectors because it's possible for data
+// from other dependencies to persist.
+const regexFormToGql = (
+  hasRegexSelectors: boolean,
+  regexForm: FormRegexSelector[]
+) =>
+  hasRegexSelectors
+    ? regexForm?.map((r) => ({
+        type: r.regexSelect,
+        data: r.regexInput,
+      }))
+    : [];
+
+const webhookFormToGql = (webhookInput: Notification["webhookInput"]) => {
+  if (!webhookInput) {
+    return null;
+  }
+  return {
+    url: webhookInput.urlInput,
+    secret: webhookInput.secretInput,
+    headers: webhookInput.httpHeaders?.map(({ keyInput, valueInput }) => ({
+      key: keyInput,
+      value: valueInput,
+    })),
+  };
+};
+const jiraFormToGql = (jiraInput: Notification["jiraIssueInput"]) => {
+  if (!jiraInput) {
+    return null;
+  }
+  return {
+    project: jiraInput.projectInput,
+    issueType: jiraInput.issueInput,
+  };
+};
+
+// Converts the form extraFields into the proper format for GQL payload.
+// We need to check what extraFields exist for a particular trigger because it's possible
+// for data from other dependencies to persist.
+const extraFieldsFormToGql = (
+  extraFieldsToInclude: ExtraField[],
+  extraFieldsForm: FormExtraFields
+) => {
+  // If there are no extra fields for this trigger, just return.
+  if (!extraFieldsToInclude) {
+    return {};
+  }
+  const extraFields = {};
+  extraFieldsToInclude.forEach((e) => {
+    extraFields[e.key] = extraFieldsForm[e.key].toString();
+  });
+  return extraFields;
+};
+
+export const getGqlPayload = (
+  subscription: Unpacked<FormState["subscriptions"]>
+): SubscriptionInput => {
+  const { subscriptionData } = subscription;
+  const event = projectTriggers[subscriptionData.event.eventSelect];
+  const { resourceType, trigger, extraFields, regexSelectors } = event || {
+    resourceType: "",
+  };
+
+  const triggerData = extraFieldsFormToGql(
+    extraFields,
+    subscriptionData.event.extraFields
+  );
+  const selectors = Object.entries(triggerData).map(([key, value]) => ({
+    type: key,
+    data: value.toString(),
+  }));
+  const regexData = regexFormToGql(
+    !!regexSelectors,
+    subscriptionData.event.regexSelector
+  );
+
+  const method = subscriptionData.notification.notificationSelect;
+  const subscriber = getTargetForMethod(method, subscriptionData?.notification);
+  return {
+    trigger,
+    resource_type: resourceType,
+    selectors: [
+      { type: "project", data: resourceType.toLowerCase() },
+      ...selectors,
+    ],
+    trigger_data: triggerData,
+    regex_selectors: regexData,
```

Comment:
I got this error:
> There was an error saving the project: saving subscriptions for project 'evergreen': 400 (Bad Request): converting subscription to service model: setting filter from selectors: setting field from selector: unknown selector type 'test-regex'

When trying to submit this form:
<img width="476" alt="image" src="https://user-images.githubusercontent.com/9298431/178302227-326fcfc6-435e-4e5a-86c2-d086e56b664c.png">

I think it may be because `test-regex` should go under regex_selectors instead of selectors

Reply:
`test-regex` needed to be passed into `trigger_data`. Its odd since this should behave the same way as the legacy  notification modal but there is likely custom parsing for some fields. I added an `allowedSelectors` list to help us filter out what elements should be passed into `selectors`.